src/main/java/core/basesyntax/service/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidOperationException;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void completeOperation(FruitTransaction fruitTransaction) {
+        Integer oldQuantity = Storage.storage.get(fruitTransaction.getName());
+        isValidQuantityToPurchase(oldQuantity, fruitTransaction.getQuantity());
+        Integer newQuantity = oldQuantity - fruitTransaction.getQuantity();
+
+        Storage.storage.put(fruitTransaction.getName(), newQuantity);
+    }
+
+    private boolean isValidQuantityToPurchase(Integer currentQuantity, Integer requiredQuantity) {
```

Comment:
You dont use boolean wich you return. It is possible to just throw exception without returning anything